near-contract-standards/src/non_fungible_token/payout/mod.rs

```diff
@@ -0,0 +1,111 @@
+mod payout_impl;
+
+use near_sdk::borsh::{BorshDeserialize, BorshSerialize};
+use near_sdk::collections::TreeMap;
+use near_sdk::json_types::U128;
+use near_sdk::AccountId;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+type BasisPoint = u16;
+
+/// This struct represents NFT royalty payout for each receiver.
+#[derive(Default, BorshDeserialize, BorshSerialize, Serialize, Deserialize)]
+#[serde(crate = "near_sdk::serde")]
+#[borsh(crate = "near_sdk::borsh")]
+pub struct Payout {
+    pub payout: HashMap<AccountId, U128>,
+}
+
+/// This struct represents percentage of total royalty per receiver as well as the total percentage
+/// of distributed royalties based on incoming payment.
+#[derive(BorshDeserialize, BorshSerialize, Debug)]
+#[borsh(crate = "near_sdk::borsh")]
+pub struct Royalties {
+    /// A mapping of accounts to the percentage of total royalty to be distributed.
+    pub accounts: TreeMap<AccountId, BasisPoint>,
+}
```

Comment:
Weeeeell, I went through another round of review and noticed a MAJOR issue here - the royalties as of this implementation are applied to all the tokens on the contract. The original idea of payouts was that each individual NFT would have its own royalties setup, e.g. every previous owner of a specific NFT could receive % of the reselling price.

Reply:
This has been fixed.